src/main/java/pt/tqs/g205/services/EncomendaService.java

```diff
@@ -0,0 +1,190 @@
+package pt.tqs.g205.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import pt.tqs.g205.domain.Cliente;
+import pt.tqs.g205.domain.Encomenda;
+import pt.tqs.g205.domain.EstadoEncomenda;
+import pt.tqs.g205.domain.EstadoEncomendaHora;
+import pt.tqs.g205.domain.Prato;
+import pt.tqs.g205.domain.PratosPorEncomenda;
+import pt.tqs.g205.domain.TipoEntrega;
+import pt.tqs.g205.repositories.ClienteRepository;
+import pt.tqs.g205.repositories.EncomendaRepository;
+import pt.tqs.g205.repositories.EstadoEncomendaHoraRepository;
+import pt.tqs.g205.repositories.EstadoEncomendaRepository;
+import pt.tqs.g205.repositories.PratoRepository;
+import pt.tqs.g205.repositories.PratosPorEncomendaRepository;
+import pt.tqs.g205.repositories.TipoEntregaRepository;
+import pt.tqs.g205.resources.models.EncomendaModel;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Service
+public class EncomendaService {
+
+  @Autowired
+  private EncomendaRepository encomendaRepo;
+
+  @Autowired
+  private ClienteRepository clienteRepo;
+
+  @Autowired
+  private ClienteService clienteService;
+
+  @Autowired
+  private TipoEntregaService tipoEntregaService;
+
+  @Autowired
+  private TipoEntregaRepository tipoEntregaRepo;
+
+  @Autowired
+  private PratoRepository pratoRepo;
+
+  @Autowired
+  private PratoService pratoService;
+
+  @Autowired
+  private EstadoEncomendaService estadoEncomendaService;
+  
+  @Autowired
+  private EncomendaRestauranteService encomendaRestauranteService;
+
+  @Autowired
+  private EstadoEncomendaRepository estadoEncomendaRepo;
+
+  @Autowired
+  private EstadoEncomendaHoraRepository eehRepo;
+
+  @Autowired
+  private PratosPorEncomendaRepository ppeRepo;
+
+  /**
+   * Obter a lista de todas as encomendas.
+   * @return lista de encomendas.
+   */
+  public List<Encomenda> getAll() {
+    List<Encomenda> encomendas = encomendaRepo.findAll();
+
+    encomendas.forEach(e -> {
+      Iterator<PratosPorEncomenda> it = e.getPratos().iterator();
+
+      while (it.hasNext()) {
+        Prato p = it.next().getPrato();
+        p.setIngredientes(null);
+        p.setRestaurante(null);
+
+      }
+    });
+
+    return encomendas;
+  }
+
+  /**
+   * Obter as encomendas de um cliente.
+   * @param clienteId id do cliente.
+   * @return lista das encomendas do cliente.
+   * @throws NoSuchElementException se o cliente nao existir.
+   */
+  public List<Encomenda> getByClienteId(Integer clienteId) throws NoSuchElementException {
+    List<Encomenda> encomendas = encomendaRepo.getByClienteId(clienteId);
+
+    if (encomendas == null || encomendas.size() == 0)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use isEmpty() to check whether the collection is empty or not. ([squid:S1155](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1155))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=20&issues=AWPIIg0QkDisRwWBeUgA&open=AWPIIg0QkDisRwWBeUgA)

[ISSUE_KEY]: AWPIIg0QkDisRwWBeUgA